### PR TITLE
bugfix/13293-export-table-data

### PR DESCRIPTION
**Pull request by Basts**
Fixed, #13293, adding x value in export data-table if the point has defined name and xAxis is not a category type.

### CODE REVIEW

File: js/modules/export-data.src.js [hunk 1]
```diff
@@ -354,12 +354,26 @@ Highcharts.Chart.prototype.getDataRows = function (multiLevelHeaders) {
             categoryMap: categoryMap,
             dateTimeValueAxisMap: dateTimeValueAxisMap
         };
+    }, 
+    // Create point array depends if xAxis is category
+    // or point.name is defined #13293
+    getPointArray = function (series, xAxis) {
+        if (series.data.filter(function (d) { return d.name; }).length &&
```
Comment (sebastianbochan) [inline js/modules/export-data.src.js:361]: I don't think that series.data.filter should be as a part of condition. Its unreadable.

I suggest to extract the filter into extra variable i.e. `namedPoints` and then use "namedPoints.length"

File: js/modules/export-data.src.js [hunk 2]
```diff
@@ -354,12 +354,26 @@ Highcharts.Chart.prototype.getDataRows = function (multiLevelHeaders) {
...
+            xAxis && !xAxis.categories &&
+            series.type !== 'gantt') {
+            if (series.pointArrayMap && series.pointArrayMap.filter(function (p) { return p === 'x'; }).length) {
```
Comment (sebastianbochan) [inline js/modules/export-data.src.js:364]: The same situation like above (line 361)

File: ts/modules/export-data.src.ts [hunk 3]
```diff
@@ -523,6 +523,25 @@ Highcharts.Chart.prototype.getDataRows = function (
...
+        // Create point array depends if xAxis is category
+        // or point.name is defined #13293
+        getPointArray = function (
+            series: Highcharts.Series,
+            xAxis: Highcharts.Axis
+        ): string[] {
+            if (
+                series.data.filter((d: Highcharts.Point): string | undefined => d.name).length &&
+                xAxis && !xAxis.categories &&
+                series.type !== 'gantt'
```
Comment (pawelfus) [inline ts/modules/export-data.src.ts:535]: Avoid checking `series.type`. Instead use `series.is('series-type')`. 

But why, in the first place, we need to filter out `gantt` type? How about x-range?

Comment (Basts) [reply ts/modules/export-data.src.ts:535]: `Gantt` chart has defined properly the `series.pointArrayMap` - (`start, end`) and doesn't require any additional changes like adding the 'x' to this array.

**Comment by TorsteinHonsi** [comment]
Please fix the commented cases and merge the master into this PR to update it.

**Pull request merged**